PROTO_tests/tests/client_dtypes_test.py

```diff
@@ -0,0 +1,163 @@
+            "Bit1//----//Bit3//----// (5)",
+            "----//----//----//Bit4// (8)",
+            "----//----//Bit3//Bit4// (12)",
+        ]
+        assert f.to_list() == expected
+
+        values = ak.arange(8, dtype=ak.uint64)
+        names = [f"Bit{x}" for x in range(65)]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", "t"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", ""]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["abc", "123"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, separator="abc")
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, pad="|!~", separator="//")
+
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_ipv4_creation(self, int_types):
+        ip_list = ak.array(IPS, dtype=int_types)
+        ipv4 = ak.IPv4(ip_list)
+
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == [format(ipaddress.IPv4Address(ip)) for ip in IPS]
+        assert ipv4.dtype == ak.bitType
+
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(f"{IPS}")
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(ak.array([IPS + 0.177]))
+
+        # Test handling of python dotted-quad input
+        ipv4 = ak.ip_address([format(ipaddress.IPv4Address(ip)) for ip in IPS])
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == [format(ipaddress.IPv4Address(ip)) for ip in IPS]
+        assert ipv4.dtype == ak.bitType
+
+    def test_ipv4_normalization(self):
+        ip_list = ak.array(IPS)
+        ipv4 = ak.IPv4(ip_list)
+        ip_as_dot = [ipaddress.IPv4Address(ip) for ip in IPS]
+        ip_as_int = [ipv4.normalize(ipd) for ipd in ip_as_dot]
+        assert IPS == ip_as_int
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_is_ipv4(self, size):
+        x = [random.getrandbits(32) for i in range(size)]
+
+        ans = ak.is_ipv4(ak.array(x, dtype=ak.uint64))
+        assert ans.to_list() == [True] * size
+
+        ipv4 = ak.IPv4(ak.array(x))
+        assert ak.is_ipv4(ipv4).to_list() == [True] * size
+
+        x = [random.getrandbits(64) if i < size / 2 else random.getrandbits(32) for i in range(size)]
+        ans = ak.is_ipv4(ak.array(x, ak.uint64))
+        assert ans.to_list() == [i >= size / 2 for i in range(size)]
+
+        with pytest.raises(TypeError):
+            ak.is_ipv4(ak.array(x, ak.float64))
+
+        with pytest.raises(RuntimeError):
+            ak.is_ipv4(ak.array(x, dtype=ak.uint64), ak.arange(2, dtype=ak.uint64))
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_is_ipv6(self, size):
+        x = [random.getrandbits(128) for i in range(size)]
```

Comment:
since `i` is never used, python devs would normally name it `_` to indicate that, so
```suggestion
        x = [random.getrandbits(128) for _ in range(size)]
```

Reply:
Updated